### PR TITLE
Fixes #39: adds a navigation drawer to the main screen

### LINKED ISSUE #39: Add Main Screen UI Layout

### PR DESCRIPTION
**Pull request by oa2013**
Fixes #39  .

Changes proposed in this pull request:
- adds a navigation drawer to the main screen

### CODE REVIEW

**Comment by oa2013** [comment]
@dynamite8 @dambadipudi main screen UI is done, needs to be integrated with resource adapter; please see the Home Fragment -- right now when you search, you will see a Toast message

**Comment by dambadipudi** [comment]
@oa2013 I commend you on making such a big commit! 

I have a question though. I do not understand the use of the Resource tab in the bottom navigation toolbar. The Resource List Fragment /Activity requires the category in order to get all resource URLs of that category to display. 

Did you add it so I could access the Fragment while you work on the ResourceAdapter?

**Comment by oa2013** [comment]
@dambadipudi Ah...I thought the Resource List needed to have its own fragment to display all available resources in case the user doesn't want to search and wants to see everything at the same time. Do you think it's a good idea to display all?

If not, we can rewrite the Resource Fragment to show up when the user searches only and remove it from the bottom nav bar.

**Comment by dambadipudi** [comment]
@oa2013 We could display all available resources on clicking on the Resource tab 👍

File: app/src/main/java/com/tiptopgoodstudio/androidresources/HomeFragment.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import android.widget.SearchView;
+import android.widget.Toast;
+
+/**
+ * Updated on 03/20/2018 by Olga Agafonova.
+ * The onQueryTextSubmit method is passing the user string to the ResourceListAdapter
+ */
+public class HomeFragment extends Fragment {
+
+    private static final String TAG = "Home";
```
Comment (dynamite8) [inline app/src/main/java/com/tiptopgoodstudio/androidresources/HomeFragment.java]: We can use the class name as the TAG.  This way if we make any name changes, we don't have to update the tag.
`private static final String TAG = HomeFragment.class.getSimpleName();`

File: app/src/main/java/com/tiptopgoodstudio/androidresources/MainActivity.java [hunk 2]
```diff
@@ -1,26 +1,111 @@
...
+        mBottomNavigationView = (BottomNavigationView) findViewById(R.id.bottom_navigation);
+
+        mBottomNavigationView.setOnNavigationItemSelectedListener(
+                new BottomNavigationView.OnNavigationItemSelectedListener() {
+                    @Override
+                    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
+                        switch (item.getItemId()) {
+                            case R.id.home:
+                                mViewPager.setCurrentItem(0);
+
```
Comment (dynamite8) [inline app/src/main/java/com/tiptopgoodstudio/androidresources/MainActivity.java]: Remove line as not needed before break;  Same comment for the other ones as well.

File: app/src/main/java/com/tiptopgoodstudio/androidresources/MainActivity.java [hunk 3]
```diff
@@ -1,26 +1,111 @@
...
+            @Override
+            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
+            }
+
+            @Override
+            public void onPageSelected(int position) {
+                if (prevMenuItem != null) {
+                    prevMenuItem.setChecked(false);
+                }
+                else
```
Comment (dynamite8) [inline app/src/main/java/com/tiptopgoodstudio/androidresources/MainActivity.java]: else on the same line as } 

`} else {`

File: app/src/main/java/com/tiptopgoodstudio/androidresources/ResourcesFragment.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+
+/**
+ * A simple {@link Fragment} subclass.
+ */
+public class ResourcesFragment extends Fragment {
+
+    private static final String TAG = "Resources";
```
Comment (dynamite8) [inline app/src/main/java/com/tiptopgoodstudio/androidresources/ResourcesFragment.java]: Use the class name

File: app/src/main/java/com/tiptopgoodstudio/androidresources/ResourcesFragment.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        // This progress bar is displayed while loading the results
+        mLoadingIndicator = (ProgressBar) view.findViewById(R.id.pb_loading_indicator);
+
+        // This TextView is used to display errors and will be hidden if there are no errors
+        mErrorMessageDisplay = (TextView) view.findViewById(R.id.tv_error_message_display);
+
+        //Now load the resources data
+        loadResourcesData();
+
+        return view;
```
Comment (dynamite8) [inline app/src/main/java/com/tiptopgoodstudio/androidresources/ResourcesFragment.java]: I like that you have comments to explain what you're doing 👍

File: app/src/main/res/drawable/ic_bigsticky.xml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+<item android:offset="0.2363" android:color="#FFFFF288" />
+<item android:offset="0.2962" android:color="#FFFBED79" />
+<item android:offset="0.3969" android:color="#FFF7E667" />
+<item android:offset="0.5167" android:color="#FFF3E15A" />
+<item android:offset="0.6728" android:color="#FFF2DF52" />
+<item android:offset="1.0" android:color="#FFF1DE50" />
+            </gradient></aapt:attr></path>
+    <path
+        android:pathData="M311.8,32.4l77.1,-15.6l16,78.9l-77.1,15.6z"
+        android:fillColor="#FFE9A4"/>
```
Comment (dynamite8) [inline app/src/main/res/drawable/ic_bigsticky.xml]: I don't understand this completely so this is my guess.  Correct me if I'm wrong.  Seems like we have hardcoded the coordinates so wondering if they're flexible to fit the strings inside the sticky notes?

File: app/src/main/res/values/colors.xml [hunk 7]
```diff
@@ -4,10 +4,15 @@
     <color name="colorPrimary">#3F51B5</color>
     <color name="colorPrimaryDark">#303F9F</color>
     <color name="colorAccent">#FF4081</color>
+    <color name="colorMenuItem">#FFFFFF</color>
```
Comment (dynamite8) [inline app/src/main/res/values/colors.xml]: Great that some of the good practices in our app. 👍

File: app/src/main/res/values/colors.xml [hunk 8]
```diff
@@ -4,10 +4,15 @@
...
 
     <!-- Colors used in resource_item.xml (Added by Divya on 3/3/2018) -->
     <color name="row_background_color">#FFFFFFFF</color>
 
     <!-- Colors used in resource_item.xml (Added by Divya on 3/3/2018) -->
     <color name="list_background_color">#FF555555</color>
+
+    <!-- Strings used in MainActivity (Added by Olga on 3/20/2018) -->
+    <color name="search_background_color">#d6d3d3</color>
```
Comment (dynamite8) [inline app/src/main/res/values/colors.xml]: Make caps for the color value to be consistent with the others?

File: app/.idea/misc.xml [hunk 9]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
[the remark this answers is not in the document]
Comment (oa2013) [reply app/.idea/misc.xml]: Yes, deleting .idea/ and app/.idea in the next commit.

File: app/src/main/java/com/tiptopgoodstudio/androidresources/HomeFragment.java (hunk 1 above)
Comment (oa2013) [reply app/src/main/java/com/tiptopgoodstudio/androidresources/HomeFragment.java]: Got it, made change in the coming commit.

File: app/src/main/java/com/tiptopgoodstudio/androidresources/MainActivity.java (hunk 2 above)
Comment (oa2013) [reply app/src/main/java/com/tiptopgoodstudio/androidresources/MainActivity.java]: Got it, made change in the coming commit.

File: app/src/main/java/com/tiptopgoodstudio/androidresources/MainActivity.java (hunk 3 above)
Comment (oa2013) [reply app/src/main/java/com/tiptopgoodstudio/androidresources/MainActivity.java]: Got it, made change in the coming commit.

File: app/src/main/java/com/tiptopgoodstudio/androidresources/ResourcesFragment.java (hunk 4 above)
Comment (oa2013) [reply app/src/main/java/com/tiptopgoodstudio/androidresources/ResourcesFragment.java]: Got it, made change in the coming commit.

File: app/src/main/res/values/colors.xml (hunk 8 above)
Comment (oa2013) [reply app/src/main/res/values/colors.xml]: Got it, made change in the coming commit.

File: app/src/main/res/drawable/ic_bigsticky.xml (hunk 6 above)
Comment (oa2013) [reply app/src/main/res/drawable/ic_bigsticky.xml]: Hey @dynamite8, I think the android:pathData coordinates depend on android:width and android:height variables and I don't know if I can set these dynamically relative to string length.

The gradient coordinates...these were automagically created by Adobe Illustator and then exported to Android Studio. I suspect that they also depend on the original width and height. 

This here is the clearest explanation of android:pathData that I could find:  [](https://medium.com/@ali.muzaffar/understanding-vectordrawable-pathdata-commands-in-android-d56a6054610e)

**Comment by oa2013** [comment]
@dynamite8 I removed the search view bar and adjusted the constraints in home fragment portrait and landscape layouts.

**Pull request merged**